backend/llm/composite_brain_qa.py

```diff
@@ -0,0 +1,611 @@
+import json
+from typing import Optional
+from uuid import UUID
+
+from fastapi import HTTPException
+from litellm import completion
+from llm.api_brain_qa import APIBrainQA
+from llm.knowledge_brain_qa import KnowledgeBrainQA
+from llm.qa_headless import HeadlessQA
+from logger import get_logger
+from modules.brain.entity.brain_entity import BrainEntity, BrainType
+from modules.brain.service.brain_service import BrainService
+from modules.chat.dto.chats import ChatQuestion
+from modules.chat.dto.inputs import CreateChatHistory
+from modules.chat.dto.outputs import GetChatHistoryOutput
+from modules.chat.service.chat_service import ChatService
+
+brain_service = BrainService()
+chat_service = ChatService()
+
+logger = get_logger(__name__)
+
+
+def format_brain_to_tool(brain):
+    return {
+        "type": "function",
+        "function": {
+            "name": str(brain.id),
+            "description": brain.description,
+            "parameters": {
+                "type": "object",
+                "properties": {
+                    "brain": {
+                        "type": "string",
+                        "description": "Brain details, need to check type too",
+                    },
+                    "question": {
+                        "type": "string",
+                        "description": "Question to ask the brain",
+                    },
+                },
+                "required": ["brain_id", "question"],
```

Comment:
Not `brain` instead of `brain_id` here ?

Reply:
yes